feature: add status colors

yuw006 opened the pull request:
closes #188

yuw006 commented:
I used [Resources](https://devexpress.github.io/devextreme-reactive/react/scheduler/docs/reference/resources/)

thaigillespie commented on frontend/src/pages/CalendarPage/index.js:
```diff
@@ -123,6 +123,18 @@ class CalendarPage extends React.Component {
+        instances: [
+          { id: 'complete', text: 'Laser Tag', color: 'cadetBlue' },
+          { id: 'pending', text: 'Test Event', color: 'plum' },
+          { id: 'ready', text: 'Test Event', color: 'tomato' },
```
Since text is an optional field (https://devexpress.github.io/devextreme-reactive/react/scheduler/docs/reference/resources/#resourceinstance) you don't have to include it here in these JS objects. If you do want to include text, I think doing something like `text: 'Readied Event'` and `'text: Pending Event'` would be a lot better.

godwinpang replied:
+1 here - I agree with Thai; probably would be nicer to leave out text, not sure what its purpose is here as of now

thaigillespie reviewed (changes requested):
Overall lgtm. I haven't used Resources so I'm not the best judge at whether or not you're using it well here (but the code looks right and your screenshots indicate that it works so it's good to me). Just had one small comment and I'll approve when you're done resolving it. 

Also, if possible, can you upload some photos of how the status colors of the events look with the whole calendar page in the description of your PR?

godwinpang commented on frontend/src/pages/CalendarPage/index.js:
```diff
@@ -138,36 +150,41 @@ class CalendarPage extends React.Component {
-              }
-              label='Complete'
-            />
+            {view === 'calendar' &&
```
For this block could just have 1 ternary statement? view==='calendar' can just control all of the 3 FormControlLabels.

godwinpang commented on frontend/src/pages/CalendarPage/index.js:
```diff
@@ -123,6 +123,18 @@ class CalendarPage extends React.Component {
     } = this.state;
     const { classes, history } = this.props;
 
+    // color of events based on status
```
What do you think about having this within the Calendar component instead of being in the CalendarPage? I think there's merit in doing so because I don't see CalendarPage as being responsible for controlling *how* the Calendar component should color the events.

Specifically, resources is of an object type that's directly from the dx-scheduler component. Having to define resources outside of the Calendar component tightly couples our CalendarPage to our *implementation* of the Calendar component. For example, if we wanted to switch away from using dx-scheduler in the future, we'd have to change both the CalendarPage and the Calendar component - this means that the abstraction that the Calendar component provides is a "leaky" one.

yuw006 commented:
I fixed the the resources field, and placement of the variable. Simplified the code to hide the check boxes. Here are the screenshots:
Calendar View:
![image](https://user-images.githubusercontent.com/63177486/113053368-0d88fe80-915d-11eb-84ad-09053823a8ba.png)
List View:
![image](https://user-images.githubusercontent.com/63177486/113053408-1974c080-915d-11eb-8de6-d21e88a83cc9.png)

thaigillespie commented on frontend/src/pages/CalendarPage/components/Calendar/index.js:
```diff
@@ -10,9 +10,29 @@ import {
+    instances: [
+      { id: 'complete', color: '#b3cce6' },
+      { id: 'pending', color: '#e6b3cc' },
+      { id: 'ready', color: '#f28c8c' },
```
I appreciate the change to hex colors!

thaigillespie commented on frontend/src/pages/CalendarPage/components/Calendar/index.js:
```diff
@@ -10,9 +10,29 @@ import {
+
+resources.propTypes = {
+  fieldName: PropTypes.string.isRequired,
+  instances: PropTypes.object.isRequired,
```
Based on how you defined `instances` starting from line 21, the proptypes for it should be an array of objects, and not just an object

thaigillespie replied:
Well actually, since resources is not even a prop of the Calendar component, you don't even need these propTypes definitions. Can probably just remove them. One day most of our files will be changed to TypeScript and we'll have a better time with types : )

thaigillespie reviewed (changes requested):
Overall lgtm, just one small thing that needs changing

godwinpang reviewed (approved):
lgtm - @thaigillespie waiting on ur review then @yuw006 feel free to squash and merge :)

thaigillespie reviewed (approved):
Lgtm @yuw006 merge whenever you're free :)
